packages/vite/src/node/config.ts

```diff
@@ -1859,18 +1859,20 @@ async function loadConfigFromBundledFile(
   // write it to disk, load it with native Node ESM, then delete the file.
   if (isESM) {
     const nodeModulesDir = findNearestNodeModules(path.dirname(fileName))
-    if (nodeModulesDir) {
+    const isDeno = !!(globalThis as any).Deno
```

Comment:
In my opinion, assuming that arbitrary files (not npm packages) placed in `node_modules/` will work regardless of the runtime is quite a leap compared to storing files meant to be consumed in the browser in `node_modules/.vite`.

I suppose that placing application files under `node_modules/` and executing it with a runtime is not something each runtime (or maybe a package manager) would expect so I'd say that the current design is dependent on the current Node.js behavior, and not something that can be widely assumed for every runtime.

(I understand it's not my call and it's definitely out of scope but just wanted to share my thoughts 💭)

Reply:
I think it is just a matter of perspective. The reason Vite works with deno is because deno is fairly compatible enough with Node, not because Vite tries to support all existing runtimes (not saying deno isn't supported, it's just how it currently works). From that point of view, deno lacks Node.js compatibility here, and Vite applies a patch for deno (that said, I personally think that behaving differently in node_modules is a valid choice on deno's side).

Reply:
Totally agree, it is a matter of perspective but I was just curious if this fairly uncommon approach (even from node's standards) to store modules that are executed by the host runtime in `node_modules/` is worth the benefits compared to the potential breaks that it may introduce for other runtimes.

Anyway, thanks for elaborating 👍